pageserver/src/object_repository.rs

```diff
@@ -923,15 +930,15 @@ struct ObjectVersionIter<'a> {
     buf_tag: BufferTag,
 
     /// Iterator on the current timeline.
-    current_iter: Box<dyn Iterator<Item = (Lsn, Vec<u8>)> + 'a>,
+    current_iter: Box<ObjectsIter<'a>>,
 
     /// Ancestor of the current timeline being iterated.
     ancestor_timeline: Option<ZTimelineId>,
     ancestor_lsn: Lsn,
 }
 
 impl<'a> Iterator for ObjectVersionIter<'a> {
```

Comment:
Do these two impls need named lifetimes?

Reply:
I'm guessing you meant these? Good catch, it looks like they can be elided:
```
impl Iterator for ObjectVersionIter<'_>
...
impl ObjectVersionIter<'_>
```